### PR TITLE
TE-6915 using new facade method to check table existance

### PR DESCRIPTION
**Pull request by gechetspr**
- Developer(s): @gechetspr

- Ticket: https://spryker.atlassian.net/browse/TE-6915

- Release Group: https://release.spryker.com/release-groups/view/3123

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   EventBehavior         | minor                 |  Propel                     |

-----------------------------------------

##### Change log

Improvements

- Added `PropelFacadeInterface` to dependencies.
- Adjusted `TriggerManager` in order to use new table existence check instead of try catch block.

Deprecations

- Deprecated `EventBehaviorQueryNotExistsException`.
- Deprecated `EventBehaviorQueryContainer::checkRequiredClasses()`.

### CODE REVIEW

**Review by bezpiatovs — APPROVED** [review]

**Review by stereomon — APPROVED** [review]

**Comment by dereuromark** [comment]
Can you rebase this on top of master, so that GitHubActions is used?
And we can see those results then in Release App for validation (green)?

File: src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php
Comment (asmarovydlo) [inline src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php:98]: We shouldn't silently ignore exceptions, especially if we no longer expect any (We do the check for table/query existence beforehand).
Please leave only `ConnectionException` here.

File: src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php
Comment (asmarovydlo) [inline src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php:85]: `EventBehaviorQueryNotExistsException` no longer used, and can be deprecated.
1. `Throw` annotation in the query container should be deprecated. 
2. Functionality that throws an exception in query container should be moved to protected method and deprecated.
3. The exception class should be deprecated.

File: src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php
Comment (asmarovydlo) [inline src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php:25]: Please add `@uses` link to \Orm\Zed\EventBehavior\Persistence\Map\SpyEventBehaviorEntityChangeTableMap::TABLE_NAME

**Review by asmarovydlo — CHANGES_REQUESTED** [review]
Instead of `Adjusted TriggerManager in order to use new table existence check instead of try catch block.` in changelogs please mention the facade method that impacted.

File: src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php
Comment (asmarovydlo) [inline src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php:116]:
```suggestion
    protected function checkRequiredClasses(): void
```

File: src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php
Comment (asmarovydlo) [inline src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php:114]:
```suggestion
     * @throws EventBehaviorQueryNotExistsException
     * 
     * @return void

```

**Review by asmarovydlo — CHANGES_REQUESTED** [review]
Deprecations should be mentioned in the changelogs.

**Review by asmarovydlo — APPROVED** [review]

File: src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php
Comment (gechetspr) [inline src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php:114]: Please use FQCN in docblocks

**Review by gechetspr — COMMENTED** [review]
Small change required.

Also I can't approve this PR as long as I want the one who created it
@asmarovydlo

**Review by asmarovydlo — APPROVED** [review]

File: src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php
Comment (gechetspr) [inline src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php:116]: This is very generic name for the query container. I would prefer smth like `eventBehaviorEntityClassExists`

**Review by gechetspr — COMMENTED** [review]
Can't approve because I was created PR, but all good from my side

File: composer.json
Comment (dereuromark) [inline composer.json:12]: For tests we need to use "dev-... as" here for them to run
Did you do that before? So we know all is good when releasing?

I wonder if we need to fix up the auto release process to include these aliasing changes for release.
So we have better testing beforehand.

**Pull request merged**
